internal/postgres/config_test.go

```diff
@@ -506,3 +507,25 @@ EOF
 chmod +x /tmp/pg_rewind_tde.sh`))
 	})
 }
+
+func TestReleaseIsFinal(t *testing.T) {
+	// On November 4th, 2024, PG 10 and 11 were EOL and 12-17 were supported.
+	testDate, err := time.Parse("2006-Jan-02", "2024-Nov-04")
+	assert.NilError(t, err)
+	assert.Check(t, ReleaseIsFinal(10, testDate))
+	assert.Check(t, ReleaseIsFinal(11, testDate))
+	assert.Check(t, !ReleaseIsFinal(12, testDate))
+	assert.Check(t, !ReleaseIsFinal(13, testDate))
+	assert.Check(t, !ReleaseIsFinal(14, testDate))
+	assert.Check(t, !ReleaseIsFinal(15, testDate))
+	assert.Check(t, !ReleaseIsFinal(16, testDate))
+	assert.Check(t, !ReleaseIsFinal(17, testDate))
```

Comment:
🤔 We have to test every version to get coverage. Is it better 🤷🏻 for the implementation to be more "data" than "behavior?"

(untested)
```go
// https://www.postgresql.org/support/versioning
var finalReleaseDates = map[int]time.Time{
	10: time.Date(2022, time.November+1, 10, 0, 0, 0, 0, time.UTC),
	11: time.Date(2023, time.November+1, +9, 0, 0, 0, 0, time.UTC),
…
}

// ReleaseIsFinal returns whether or not t is definitively past the final
// scheduled release of a Postgres version.
func ReleaseIsFinal(majorVersion int, t time.Time) bool {
	known, ok := finalReleaseDates[majorVersion]
	return ok && t.After(known)
}
```

Reply:
Yeah I think I like this better